integration-tests/tests/src/tests/migrations.ts

```diff
@@ -0,0 +1,521 @@
+          },
+        ],
+        schemaVersion: 1,
+        onMigration: function (oldRealm, newRealm) {
+          const oldSchema = oldRealm.schema;
+          const newSchema = newRealm.schema;
+          expect(oldSchema.length).equals(1);
+          expect(newSchema.length).equals(1);
+
+          expect(oldSchema[0].name).equals("Test");
+          expect(newSchema[0].name).equals("Test");
+
+          expect(oldSchema[0].properties.prop0.type).equals("string");
+          expect(newSchema[0].properties.prop0).equals(undefined);
+
+          expect(oldSchema[0].properties.prop1.type).equals("int");
+          expect(newSchema[0].properties.prop1.type).equals("int");
+
+          expect(oldSchema[0].properties.renamed).equals(undefined);
+          expect(newSchema[0].properties.renamed.type).equals("string");
+        },
+      });
+      migratedRealm.close();
+    });
+  });
+
+  describe("Deleting models", () => {
+    beforeEach(() => {
+      Realm.clearTestState();
+    });
+
+    let realm: Realm;
+    beforeEach(() => {
+      realm = new Realm({ schema: [TestSchema] });
+
+      realm.write(function () {
+        realm.create("Test", ["stringValue", 1]);
+      });
+
+      realm.close();
+    });
+
+    it("should removing schemas from configuration correctly", () => {
+      realm = new Realm({ schema: [], schemaVersion: 1 });
+      expect(realm.schema.length).equals(0); // no models
+      realm.close(); // this won't delete the model
+
+      realm = new Realm({ schema: [TestSchema], schemaVersion: 2 });
+      expect(realm.objects("Test").length).equals(1); // the model objects are still there
+      realm.close();
+
+      // now delete the model explicitly, which should delete the objects too
+      realm = new Realm({
+        schema: [],
+        schemaVersion: 3,
+        onMigration: function (oldRealm, newRealm) {
+          newRealm.deleteModel("Test");
+        },
+      });
+
+      expect(realm.schema.length).equals(0); // no models
+
+      realm.close();
+
+      realm = new Realm({ schema: [TestSchema], schemaVersion: 4 });
+
+      expect(realm.objects("Test").length).equals(0);
+
+      realm.close();
+    });
+
+    it("should handle deleteModel correctly", () => {
+      // now delete the model explicitly, but it should remain as it's still in the schema
+      // only the rows should get deleted
+      realm = new Realm({
+        schema: [TestSchema],
+        schemaVersion: 1,
+        onMigration: function (oldRealm, newRealm) {
+          newRealm.deleteModel("Test");
+        },
+      });
+
+      expect(realm.schema.length).equals(1); // model should remain
+      expect(realm.objects("Test").length).equals(0); // objects should be gone
+
+      realm.close();
+
+      realm = new Realm({ schema: [TestSchema], schemaVersion: 2 });
+      expect(realm.objects("Test").length).equals(0);
+
+      realm.close();
+    });
+    it("should ignore deleteModel with non-existing models", () => {
```

Comment:
Our linter never picks this up, but I would prefer new lines between `it` clauses.